tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1228,3 +1225,34 @@ def test_single_sign_on_using_rhsso(enable_external_auth_rhsso, session):
 
     finally:
         update_rhsso_settings_in_satellite(revert=True)
+
+
+@destructive
+def test_ext_logout_rhsso(session):
+    """Verify the ext logout page navigation with external authentication RH-SSO
+
+    :id: 87b5e08e-69c6-11ea-8126-e74d80ea4308
+
+    :setup: Enroll the RH-SSO Configuration for External Authentication
+
+    :steps:
+        1. Create Mappers on RHSSO Instance and Update the Settings in Satellite
+        2. Login into Satellite using RHSSO login page redirected by Satellite
+        3. Logout from Satellite and Verify the ext_logout page displayed
+
+    :expectedresults: After logout from Satellite navigate should be ext_loout page
+    """
+    try:
```

Comment:
what is the most volatile part of this code? i'd like us to wrap as little as possible under a ```try```

Reply:
@JacobCallahan can you please explain in detail here. Not sure what do you mean here.

Reply:
if you mean why I'm doing try-catch here, This is because we are changing the settings, if we don't use that can cause a radical issue on the server. hence I'm reverting the settings. Please let me know if you are looking for.

Reply:
@omkarkhatavkar i'm basically trying to find out what part of this is most likely to fail. then wrap only that line (or two) in the try

Reply:
I can write the decorator wrapper like ldap_wrapper here is e.g. https://github.com/omkarkhatavkar/robottelo/blob/b164fe01e51d6b7b2015762a9eacbc21189ac651/tests/foreman/ui/test_ldap_authentication.py#L356 if you think it is a better idea?

Reply:
so that decorator would wrap the entire test in a try/finally :)
basically, if the most volatile lines of this test are ```session.rhsso_login.login(login_details)``` those are the only ones that should be wrapped in the try/finally

Reply:
@JacobCallahan `session.rhsso_login.login(login_details, external_login=True)` line also need that try catch, if we add only on `session.rhsso_login.login(login_details)`, mentioned line will fail.

Reply:
@JacobCallahan the idea here is that we want to run cleanup no matter what happens. This is to restore global configurations to known working state, clean up external objects etc. In pytest you could model that with fixtures and finalizers. I don't know if there is better way to do that in unittest, but we kind of settled on `try/finally`, with large chunks of code inside `try`. Just grep robottelo tests for `finally:` for evidence.

It seems to me that you two kind of talk past each other, and as far as I can tell, root cause is that you have two different use-cases in mind.

Reply:
I'm open here to know the solution or hint about how I can implement. Is that something you guys can let me then it works for me.

Reply:
@JacobCallahan Do you have more concerns, I need this need to be merge before next snap. Please let me know

Reply:
@mirekdlugosz we'll have to keep this in mind during the great unittest purge. While not ideal, I won't block this time.